include/dsn/dist/distributed_lock_service.h

```diff
@@ -69,7 +69,7 @@ namespace dsn
             /*
              * initialization routine
              */
-            virtual error_code initialize() = 0;
+            virtual error_code initialize(const char* work_dir, const char* lock_root) = 0;
```

Comment:
But then user may need to config duplication values, for example, config cluster_root for meta_state_service, and config lock_root for distributed_lock_service, which is not user-friendly.